spyder/plugins/workingdirectory/container.py

```diff
@@ -77,7 +81,61 @@ def enterEvent(self, event):
         """Set current path as the tooltip of the widget on hover."""
         self.setToolTip(self.currentText())
 
+    def focusOutEvent(self, event):
+        """Handle focus out event restoring the last valid selected path."""
+        if self.add_current_text_if_valid():
+            self.selected()
+            self.hide_completer()
+        hide_status = getattr(self.lineEdit(), 'hide_status_icon', None)
+        if hide_status:
+            hide_status()
+        super().focusOutEvent(event)
+
+    # --- Own methods
```

Comment:
```suggestion
    # ---- Own methods
```

So that this section appears in our Outline.